scripts/delayed.js

```diff
@@ -35,20 +35,22 @@ loadScript('https://icici-securities.allincall.in/files/deploy/embed_chatbot_11.
 async function loadGTM() {
   const scriptTag = document.createElement('script');
   scriptTag.innerHTML = `
-        (function (w, d, s, l, i) {
-        w[l] = w[l] || [];
-        w[l].push({
-            'gtm.start':
-                new Date().getTime(), event: 'gtm.js'
-        });
-        var f = d.getElementsByTagName(s)[0],
-            j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : '';
-        j.async = true;
-        j.src =
-            'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
-        f.parentNode.insertBefore(j, f);
-        }(window, document, 'script', 'dataLayer', 'GTM-WF9LTLZ'));
+        (function (w, d, s, l, i, r) {
+    w[l] = w[l] || [];
+    w[l].push({
+        'gtm.start': new Date().getTime(),
+        event: 'gtm.js',
+        'referrer': r // Pass the original referrer here
+    });
+    var f = d.getElementsByTagName(s)[0],
+        j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : '';
+    j.async = true;
+    j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + '&referrer=' + encodeURIComponent(r);
+    console.log(j.src);
+    f.parentNode.insertBefore(j, f);
+}(window, document, 'script', 'dataLayer', 'GTM-WF9LTLZ', document.referrer));;
```

Comment:
@sdmcraft  any thought on keeping the original referrer intact even with delay ?